lib/value-pairs/transforms.c

```diff
@@ -152,6 +151,39 @@ value_pairs_new_transform_shift (gint amount)
   return (ValuePairsTransform *)vpt;
 }
 
+/* shift-levels() */
+
+static void
+vp_trans_shift_levels(ValuePairsTransform *t, GString *key)
+{
+  VPTransShift *self = (VPTransShift *)t;
+  const gchar *dot;
+  gint levels_left = self->amount - 1;
+
+  dot = strchr(key->str, '.');
```

Comment:
Now we use `LL_NUMBER` in grammar, which allows 0 or less. For those cases, we will also shift one level. Maybe it would better use `positive_integer` in the grammar.

Reply:
fixed both for --shift and --shift-levels